### DISCUSSION TITLE
Plex HDR SDR Hardware Transcoding

### DISCUSSION THREAD

**Discussion by HowMotion4512** [tteck/Proxmox#179] [General]
Hello! 

Thank you for all the work creating these helper scripts for Proxmox, they are immensely helpful and I'm so grateful that you have put these together. 

I'm running the Plex in an LXC on Proxmox, that I set up using your helper script. It works great and is able to HW transcode without issue on my 11th gen Intel NUC. However, I can't seem to figure out how to get HW transcoding working when HDR to SDR tone mapping is enabled. I understand that there are a few additional drivers required, including those listed [here](https://support.plex.tv/articles/hdr-to-sdr-tone-mapping/) and [here](https://github.com/intel/compute-runtime/releases). Even after installing those in the host and the container, HW transcoding with tone mapping enabled still doesn't work. 

Would it be possible for you to integrate a fix for this issue into the Plex helper script? 

Thank you again!!

**Comment by tteck** [comment]
```
## 2022-03-24

### Changed

- **Plex Media Server LXC**
  - Switch to Ubuntu 20.04 to support HDR tone mapping
```
This supposedly fixed the issue 
https://github.com/tteck/Proxmox/discussions/90

**Comment by tteck** [comment]
If you installed before that date, you need to reinstall and run (in the Proxmox shell)
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/pms-copy-data.sh)"
```
to copy your media over to the new LXC

**Comment by HowMotion4512** [comment]
Thank you so much, I was able to get it working!! 

In addition to using your latest script, however, I had to install version 21.49.21786 of Intel Compute Runtime. Here is a note from the Plex team saying that it's required for intel 11th gen CPUs. 

https://forums.plex.tv/t/4k-hdr-tone-mapping-proxmox-lxc-hardware-transcoding-11th-gen-intel-nuc/789093/3

**Comment by tteck** [comment]
Hopefully the update coming for the Debian packing will help

**Comment by thebigluke** [comment]
Hi, apologies for thread-jacking but it's related (I think?!)

I tried installing but it fails on Hardware Acceleration:

 - Setting Up Hardware Acceleration.../bin/chmod: cannot access '/dev/dri/*': No such file or directory
‼ ERROR 1@82 Unknown failure occured.

I have an AMD Vega 64 which I passthrough to Windows11 for daily use, and 9700k intel cpu of which I use the graphics on my 2nd monitor for the Proxmox host via hdmi.

After it fails, the new plex CT is still running and Ubuntu prompts for a username and password in the shell, which I don't know...

Any ideas on either the Ubuntu username/password or the hardware acceleration error?

Thanks

**Comment by xppx99** [comment]
Hello tteck, 

I've been struggling to enable HW transcoding when tonemapping setting is enable. 
I've just completed a new proxmox install, followed by running your (very good) script to install a Plex LXC, and I confirm that HW transcoding works except when HDR to SDR with the setting enabled. 

Note: It's implied that I also installed on the host the packages: i965-va-driver, ocl-icd-libopencl1 and beignet-opencl-icd

My Intel CPU is the model i5-7200U. 
Do you suggest any approach?

**Reply by tteck** [reply]
Are you using Ubuntu 20.04?
https://github.com/tteck/Proxmox/discussions/179#discussioncomment-2595975

**Reply by xppx99** [reply]
yes, forgot to mention.

**Reply by xppx99** [reply]
Just to clarify that after I posted the issue in [Plex Forums](https://forums.plex.tv/t/plex-server-in-lxc-proxmox-no-tonemapping-hw-transcode/820473) the situation was solved. 
There is nothing wrong with the LXC component (tteck), the issue is with Plex Server version. The Version 1.29.2.6273 works fine.
